### PR TITLE
fix: [PL-22273]: Added the log lines to check if the JWT Token is bei…

### PR DESCRIPTION
**Pull request by boopesh007**
fix: [PL-22273]: Added the log lines to check if the JWT Token is being used without Exp field.

You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

[PL-22273]: https://harness.atlassian.net/browse/PL-22273?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/30585)

### CODE REVIEW

**Comment by boopesh007** [comment]
trigger messagecheck

**Comment by boopesh007** [comment]
trigger codeformat

**Comment by boopesh007** [comment]
trigger codeformat

File: 980-commons/src/main/java/io/harness/security/JWTTokenServiceUtils.java [hunk 1]
```diff
@@ -49,6 +49,9 @@
 
   public Pair<Boolean, Map<String, Claim> > isServiceAuthorizationValid(String serviceToken, String serviceSecret) {
     Map<String, Claim> claimMap = verifyJWTToken(serviceToken, serviceSecret);
+    if (!claimMap.containsKey("exp")) {
```
Comment (karansiwach360) [inline 980-commons/src/main/java/io/harness/security/JWTTokenServiceUtils.java:52]: use this method SecurityContextBuilder.getPrincipalFromClaims to get principal from claims and if it is not null then log principal.type and principal.name as well.

Same for other places as well.

**Review by karansiwach360 — COMMENTED** [review]
> Reviewed 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30585)* status: 3 file(s) to review. 0 of 2 approvals obtained.

**Review by karansiwach360 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30585)* status: 3 file(s) to review. 1 of 2 approvals obtained.

File: 980-commons/src/test/java/io/harness/security/JWTTokenServiceUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        assertThat(authMap.getKey().booleanValue()).isTrue();
+        assertThat(authMap.getValue().containsKey("exp")).isFalse();
+        assertThat(authMap.getValue().containsKey("type")).isFalse();
+    }
+
+    @Test(expected = InvalidRequestException.class)
+    @Owner(developers = BOOPESH)
+    @Category(UnitTests.class)
+    public void isInvalidServiceAuthTestExpiredToken() throws InterruptedException {
+        ImmutableMap<String, String> claims = ImmutableMap.of("email", email);
```
Comment (karansiwach360) [inline 980-commons/src/test/java/io/harness/security/JWTTokenServiceUtilsTest.java]: Define "email", "exp" at the class level and reuse.
private static final String URL = "url";

File: 980-commons/src/test/java/io/harness/security/JWTTokenServiceUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+import java.util.Date;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.lang3.tuple.Pair;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+@OwnedBy(PL)
+public class JWTTokenServiceUtilsTest extends CategoryTest {
+  private final String ISSUER = "Harness Inc";
```
Comment (karansiwach360) [inline 980-commons/src/test/java/io/harness/security/JWTTokenServiceUtilsTest.java:32]: private static final and keep the static variable names in uppercase

**Review by karansiwach360 — COMMENTED** [review]
> Reviewed 2 of 3 files at r4, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30585)* status: 5 file(s) to review. 0 of 2 approvals obtained.

**Review by karansiwach360 — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30585)* status: 5 file(s) to review. 1 of 2 approvals obtained.

**Review by pbarapatre10 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30585)* status: 5 file(s) to review. 

___
*[120-ng-manager/src/main/java/io/harness/ng/core/invites/api/impl/InviteServiceImpl.java, line 608 at r6](https://reviewable.io/reviews/harness/harness-core/30585#-MvbdYwg2eipdIHGKX77:-MvbdYwh6SPUTPHON0Ag:b-28t5no) ([raw file](https://github.com/harness/harness-core/blob/45dd2417067e3a4ae2af2fe0f6ddb8d5ece1c66f/120-ng-manager/src/main/java/io/harness/ng/core/invites/api/impl/InviteServiceImpl.java#L608)):*
> ```Java
>       Principal principal = SecurityContextBuilder.getPrincipalFromClaims(claims);
>       if (principal != null) {
>         log.warn(String.format(
> ```

nit: I think this should be info and not warn, since we are logging just the principal info here.
Thanks
___
*[400-rest/src/main/java/software/wings/security/AuthenticationFilter.java, line 329 at r6](https://reviewable.io/reviews/harness/harness-core/30585#-MvbdmCr3U5O1IVWWaBT:-MvbdmCr3U5O1IVWWaBU:b9g9e01) ([raw file](https://github.com/harness/harness-core/blob/45dd2417067e3a4ae2af2fe0f6ddb8d5ece1c66f/400-rest/src/main/java/software/wings/security/AuthenticationFilter.java#L329)):*
> ```Java
>         Principal principal = SecurityContextBuilder.getPrincipalFromClaims(claimMap);
>         if (principal != null) {
>           log.warn(String.format(
> ```

same as above, info rather than warn here

**Review by pbarapatre10 — APPROVED** [review]
> Reviewed 1 of 1 files at r6, 4 of 4 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30585)* status: 4 file(s) to review. 1 of 2 approvals obtained.

**Review by ujjawal-harness — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r6, 4 of 4 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30585#-:-MvqGhCn4ow7eRsA_Tq4:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**